ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_linalg.py

```diff
@@ -606,3 +606,55 @@ def test_paddle_matrix_power(
         x=x[0],
         n=n,
     )
+
+
+# cond
+@st.composite
+def _get_dtype_and_matrix_non_singular(draw):
+    [None, "fro", "nuc", np.inf, -np.inf, 1, -1, 2, -2]
+    while True:
+        matrix = draw(
+            helpers.dtype_and_values(
+                available_dtypes=(
+                    ivy.float64,
+                    ivy.double,
+                ),
+                min_value=-10,
+                max_value=10,
+                min_num_dims=2,
+                max_num_dims=2,
+                min_dim_size=1,
+                max_dim_size=5,
+                shape=st.tuples(st.integers(1, 5), st.integers(1, 5)).filter(
+                    lambda x: x[0] == x[1]
+                ),
+                allow_inf=False,
+                allow_nan=False,
+            )
+        )
+        if np.linalg.det(matrix[1][0]) != 0:
+            break
+
+    return matrix[0], matrix[1]
```

Comment:
Why returning two matrices here as we only need one Tensor for the function

Reply:
> Why returning two matrices here as we only need one Tensor for the function

I think we need to return the dtypes of the input matrix and the input matrix itself. we need to calculate the condition number of the matrix for the dtypes  therefore returning one matrix for the dtype and the other matrix for the matrix itself. What do you think about this?

Reply:
Got it here, but you can rename one to be dtype as it is confusing seems like it's two matrices being generated